spacepy/irbempy/__init__.py

```diff
@@ -83,11 +83,2472 @@
+                )
+
+                # take out all the odd 'bad values' and turn them into NaN
+                if np.isclose(bmin, badval):
+                    bmin = np.nan
+                GEOcoord[np.where(np.isclose(GEOcoord, badval))] = np.nan
+                # print('bmin, GEOcoord = {0},{1}'.format(bmin, GEOcoord))
+                if not (np.isnan(bmin)):
+                    # Now get Lstar at this location...
+                    postry = spc.Coords(GEOcoord, 'GEO', 'car', use_irbem=True)
+                    LStry = get_Lstar(tt, postry, pa, extMag=extMag, options=options, omnivals=omnivals)
+                    LStry['K'] = LStry['Xj']*np.sqrt(LStry['Bmirr']*nTtoG)
+                else:
+                    LStry = {'Lstar': np.nan, 'K': np.nan}
+                if 'verbose' in kwargs:
+                    print('L* at test point: {0}; Xgsm = {1}'.format(LStry['Lstar'], pos_test))
+
+                if np.isnan(LStry['Lstar']).any():
+                    loci_brac2 = pos_test
+                else:
+                    loci_brac1 = pos_test
+                    LCDS, LCDS_K = LStry['Lstar'][0], LStry['K'][0]
+            try:
+                results['LCDS'][idxt, idxa] = LCDS
+                results['K'][idxt, idxa] = LCDS_K
+            except IndexError:
+                results['LCDS'][idxt] = LCDS
+                results['K'][idxt] = LCDS_K
+
+    results['LCDS'] = results['LCDS']
+    results['K'] = results['K']
+
+    return results
+
+
+# -----------------------------------------------
+def find_LCDS_K(ticks, K, extMag='T01STORM', options=[1, 1, 3, 0, 0], omnivals=None,
+                tol=0.05, bracket=[3, 12], mlt=0, **kwargs):
+    """
+    Find the last closed drift shell (LCDS) for a given K (modified 2nd invariant).
+
+    Uses the IRBEM library to determine L* and searches via bisection to find LCDS
+    to a given tolerance in R (radial distance along GSM equator at local midnight).
+
+    Parameters
+    ----------
+        - ticks (Ticktock class) : containing time information **for a single time**
+        - K (numeric) : K (modified 2nd adiabatic invariant) for search
+        - extMag (string) : optional; will choose the external magnetic field model
+                            possible values ['0', 'MEAD', 'T87SHORT', 'T87LONG', 'T89',
+                            'OPQUIET', 'OPDYN', 'T96', 'OSTA', 'T01QUIET', 'T01STORM',
+                            'T05', 'ALEX', 'TS07']
+        - options (optional list or array of integers length=5) : explained in Lstar
+        - omni values as dictionary (optional) : if not provided, will use lookup table
+        - (see Lstar documentation for further explanation)
+        - tol (float) : tolerance for search in radial distance
+        - bracket (list): X-GSM coordinates to bracket bisection search
+
+    Returns
+    -------
+        results (SpaceData, dictionary-like): contains keys LCDS, K, AlphaEq and UTC
```

Comment:
I am not going to comment on every dosctring that is like this, but basically all of the docstrings in this file should be looked at and touched up.